netket/sampler/exact.py

```diff
@@ -126,4 +127,11 @@ def _sample_chain(
             sampler.n_chains_per_rank, chain_length, sampler.hilbert.size
         )
 
+        # TODO run the part above in parallel
+        if config.netket_experimental_sharding:
+            samples = jax.lax.with_sharding_constraint(
+                samples,
+                jax.sharding.PositionalSharding(jax.devices()).reshape(1, -1, 1),
```

Comment:
with mpi and exactsampler we have 1 fake chain per rank. Here I don't need to to this and instead I have 1 global chain (axis 0 of size 1), whose samples are sharded (axis 1 of size Ns/n_devices on each device) . Since we flatten the samples later, this gives the same as we would have had with a separate fake chain per device

Reply:
can you add this as a comment?